src/main/java/teammates/googleSendgridJava/Sendgrid.java

```diff
@@ -0,0 +1,755 @@
+package teammates.googleSendgridJava;
+
+import java.net.HttpURLConnection;
+import java.util.*;
+import java.io.IOException;
+import java.util.Iterator;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.UnsupportedEncodingException;
+
+import com.google.appengine.labs.repackaged.org.json.JSONException;
+import com.google.appengine.labs.repackaged.org.json.JSONObject;
+import com.google.appengine.labs.repackaged.org.json.JSONArray;
+
+
+public class Sendgrid {
+    
+    private String from,
+                   from_name,
+                   reply_to,
+                   subject,
+                   text,
+                   html;
+    protected Boolean use_headers = true;
+    private String serverResponse = "";
+    private ArrayList<String> to_list  = new ArrayList<String>();
+    private ArrayList<String> to_name_list  = new ArrayList<String>();
+    private ArrayList<String> bcc_list = new ArrayList<String>();
+    private JSONObject header_list = new JSONObject();
+
+    protected String domain = "https://sendgrid.com/",
+                     endpoint= "api/mail.send.json",
+                     username,
+                     password;
+
+    public Sendgrid(String username, String password) {
+        this.username = username;
+        this.password = password;
+        try {
+            this.setCategory("google_sendgrid_java_lib");
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * getTos - Return the list of recipients
+     *
+     * @return  List of recipients
+     */
+    public ArrayList<String> getTos() {
+        return this.to_list;
+    }
+
+    /**
+     * setTo - Initialize a single email for the recipient 'to' field
+     * Destroy previous recipient 'to' data.
+     *
+     * @param    email   A list of email addresses
+     * @return           The SendGrid object.
+     */
+    public Sendgrid setTo(String email) {
+        this.to_list = new ArrayList<String>();
+        this.addTo(email);
+
+        return this;
+    }
+
+    /**
+     * addTo - Append an email address to the existing list of addresses
+     * Preserve previous recipient 'to' data.
+     *
+     * @param    email   Recipient email address
+     * @param    name    Recipient name
+     * @return           The SendGrid object.
+     */
+    public Sendgrid addTo(String email, String name) {
+        if (this._useHeaders() == true){
+            String toAddress = (name.length() > 0) ? name + "<" + email + ">" : email;
+            this.to_list.add(toAddress);
+        } else {
+            if (name.length() > 0){
+              this._addToName(name);
+            } else {
+                this._addToName("");
+            }
```

Comment:
`this._addToName(name.length() > 0 ? name : "")` ?
Anyway, I don't see the point of this check. Isn't it same as `this._addToName(name)`?